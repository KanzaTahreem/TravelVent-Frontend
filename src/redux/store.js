import { configureStore } from '@reduxjs/toolkit';
import cartItems from './cartSlice';
import registrationSlice from './reducers/registrationSlice';

const store = configureStore({
  reducer: {
    cartItems,
    registration: registrationSlice,
  },
});

export default store;
