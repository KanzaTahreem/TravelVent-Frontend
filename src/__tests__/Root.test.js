import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import RootLayout from '../components/Root';

describe('RootLayout', () => {
  let root;
  beforeAll(async () => {
    root = renderer
      .create(
        <Provider store={store}>
          <Router>
            <RootLayout />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  it('should renders correctly', () => {
    expect(root).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(root).toMatchSnapshot();
  });
  it('renders desktop navigation menu items', () => {
    waitFor(() => expect(getByText('Home')).toBeInTheDocument());
    waitFor(() => expect(getByText('Add New Trip')).toBeInTheDocument());
    waitFor(() => expect(getByText('My Reservations')).toBeInTheDocument());
    waitFor(() => expect(getByText('Delete a Trip')).toBeInTheDocument());
    waitFor(() => expect(getByText('Reserve a Trip')).toBeInTheDocument());
  });
});
