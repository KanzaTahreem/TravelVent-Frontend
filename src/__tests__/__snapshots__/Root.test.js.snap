// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootLayout should match the snapshot 1`] = `
<div
  className="root-layout"
>
  <nav
    className="desktop-nav column"
  >
    <img
      alt="app-logo"
      className="desk-nav-logo"
      src="logo.png"
    />
    <ul
      className="column"
    >
      <li>
        <a
          className=""
          href="/home"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className=""
          href="/add-trip"
          onClick={[Function]}
        >
          Add New Trip
        </a>
      </li>
      <li>
        <a
          className=""
          href="/my-reservations"
          onClick={[Function]}
        >
          My Reservations
        </a>
      </li>
      <li>
        <a
          className=""
          href="/delete-trip"
          onClick={[Function]}
        >
          Delete a Trip
        </a>
      </li>
      <li>
        <a
          className=""
          href="/reserve/0"
          onClick={[Function]}
        >
          Reserve a Trip
        </a>
      </li>
    </ul>
    <div
      className="desk-copy-right column"
    >
      <p
        className="footer"
      >
        Copyright Â© 2023
        <br />
        <a
          href="https://github.com/KanzaTahreem"
          rel="noopener noreferrer"
          target="_blank"
        >
          Kanza
        </a>
        , 
        <a
          href="https://github.com/shakerAbuDrais"
          rel="noopener noreferrer"
          target="_blank"
        >
          Shaker
        </a>
         & 
        <a
          href="https://github.com/Mhamad-Raad/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Mhamad
        </a>
      </p>
    </div>
  </nav>
  <nav
    className="mobile-nav column"
  >
    <button
      className="hamburger column"
      onClick={[Function]}
      type="button"
    >
      <div
        className="line"
      />
      <div
        className="line"
      />
      <div
        className="line"
      />
    </button>
  </nav>
</div>
`;
